typescript/big-practice/src/scripts/controllers/noteController.ts

```diff
@@ -35,47 +35,41 @@ export default class NoteController {
 
   renderTabs() {
     const handlers = {
-      renderTabNotes: () => this.renderTabNote(),
-      renderTabTrash: () => this.renderTabTrash(),
+      renderTabNotes: () => this.renderTab('listNotes'),
+      renderTabTrash: () => this.renderTab('trashNotes'),
       addNote: (note: Note) => this.addNote(note),
     };
 
     this.view.renderTabs(handlers);
   }
 
-  async renderTabTrash() {
+  async renderTab(tab: string) {
     try {
       this.loadingPage.addLoading();
-      const listTrash: Note[] = await this.model.filterNotes('trashNotes');
-      // function render trash notes
-      this.view.renderListTrashNotes(listTrash, (noteId: string) =>
-        this.handleConfirmPopup(noteId)
-      );
-
-      // function show Empty Note if note is empty
-      this.view.showHideEmpty(listTrash, 'trashNotes');
-      this.loadingPage.setTimeoutLoading();
-    } catch (error) {
-      if (error instanceof Error) {
-        renderPopupError(error.message);
-      }
-    }
-  }
 
-  async renderTabNote() {
-    try {
-      this.loadingPage.addLoading();
-      const handlers = {
-        handleDeleteNote: (id: string) => this.deleteNote(id),
-        handleShowNoteForm: (id: string) => this.handleNoteForm(id),
-      };
-      const listNotes: Note[] = await this.model.filterNotes('listNotes');
-
-      // function render list notes
-      this.view.renderListNotes(listNotes, handlers);
+      if (tab === 'trashNotes') {
+        const listTrash: Note[] = await this.model.filterNotes('trashNotes');
+        // function render trash notes
+        this.view.renderListTrashNotes(listTrash, (noteId: string) =>
+          this.handleConfirmPopup(noteId)
+        );
+
+        // function show Empty Note if note is empty
+        this.view.showHideEmpty(listTrash, 'trashNotes');
```

Comment:
both should reuse tab name
`const listTrash: Note[] = await this.model.filterNotes(tab);`
`this.view.showHideEmpty(listTrash, tab);` instead